### PR TITLE
Adding tests for Uri and UriParser

### LINKED ISSUE #12156: Add tests for System.Uri and friends
On PR #11988 we added the implementation of System.Uri and friends, but we didn't add tests. The reason for this is that we want to using Mono tests and it is still not sure on how they should be consumed. This issue is to track the work of adding them once we figure out the right process.

### PR DESCRIPTION
**Pull request by joperezr**
Fixes #12156 

Adding tests for the recent added classes System.Uri and UriParser

cc: @AlexGhiondea for copyright headers since all UriParser tests are ported from mono.

cc: @danmosemsft @CIPop

### CODE REVIEW

**Comment by davidsh** [comment]
cc: @stephentoub for tests review, @richlander for verifying additional copyright headers

File: src/System.Private.Uri/tests/FunctionalTests/UriParserTest.cs [hunk 1]
```diff
@@ -0,0 +1,556 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// See the LICENSE file in the project root for more information.
+// Author:
```
Comment (AlexGhiondea) [inline src/System.Private.Uri/tests/FunctionalTests/UriParserTest.cs]: If you have made significant changes to the tests, you need to add this line:

// Copyright (c) .NET Foundation and Contributors

Comment (joperezr) [inline src/System.Private.Uri/tests/FunctionalTests/UriParserTest.cs]: mm all I changed in the tests themselves was test framework related, so basically porting them to xunit, but functioning-wise I didn't change a thing.

Comment (stephentoub) [inline src/System.Private.Uri/tests/FunctionalTests/UriParserTest.cs]: Do we need to keep the "author" information, or just the copyright section?  Seems like the author information is logically a separate comment that just happens to be at the top.

**Comment by joperezr** [comment]
Merging now since I had a green build before rebasing, and all I changed was to add an extra empty line to separate license headers.

**Pull request merged**